ShimmerDriver/src/main/java/com/shimmerresearch/driver/ShimmerDevice.java

```diff
@@ -2107,9 +2107,11 @@ public boolean isSupportedSdLogSync() {
 	}
 	
 	public boolean isHWAndFWSupportedBtBleControl() {
-		if(getFirmwareIdentifier()==ShimmerVerDetails.FW_ID.LOGANDSTREAM
+		if((getFirmwareIdentifier()==ShimmerVerDetails.FW_ID.LOGANDSTREAM
 				&& mShimmerVerObject.compareVersions(HW_ID.SHIMMER_3, FW_ID.LOGANDSTREAM, 1, 0, 4)
-						&& isShimmerGen3()){
+						&& isShimmerGen3()
```

Comment:
end bracket misplaced here with one missing after isShimmerGen3()? 

Suggested update for easier to read:

		if((isShimmerGen3() && getFirmwareIdentifier()==ShimmerVerDetails.FW_ID.LOGANDSTREAM
				&& mShimmerVerObject.compareVersions(HW_ID.SHIMMER_3, FW_ID.LOGANDSTREAM, 1, 0, 4))
			|| (isShimmerGen3R() && getFirmwareIdentifier()==ShimmerVerDetails.FW_ID.LOGANDSTREAM
					&& mShimmerVerObject.compareVersions(HW_ID.SHIMMER_3R, FW_ID.LOGANDSTREAM, 1, 0, 39) 		)){
			return true;
		}